README.md
# The Minimilistic Open Source Dashboard💜

I developed this app/chrome extension as a side project because I was fascinated with the chrome extension "Momentum" and I wanted to test out my skills to see if I would be able to make something that I can use in my daily life and Viola after a week and a lot stack overflow, here's what I came up with😄.

This current version is the basic version of the app with the basic features like to-do, greetings, weather and customizeable background images. I'll be working on making it the best free dashboard available. If you are a fellow developer, I'd love any contributions that you'd provide to this repository, for the main goal of this dashboard is promoting Open Source🥳

## Technologies Used**

* ReactJS - For frontend
* UseContextAPI - State management
* UnsplashAPI - For background images
* Free Weather API - For real time weather
* TailwindCSS - For styling the app
* Local storage - Used as a database
* A Hell Load Of Coffee☕ - What every programmer needs to stay awake😎

## Setting Up The App

**If you want a quick preview of the app you can directly visit this [link](https://open-dashboard.vercel.app/) (Reload the page, if you see null value in name, It's a bug🪲)**

### To use the chrome extension

* Download the zipped build from the releases folder
* Extract the file, you'll land up with a build folder
* Go to chrome -> extensions -> developer mode -> add unpacked
* Select the build folder and add it
* Open a new tab and see your very own dashboard😄

  **If you don't get the hello screen, try to reload the page, you'll reach the hello page🪲

### To setup the app in local system

* Fork the repository
* Clone the repository to your local system
* run `npm i` to install the dependencies
* After all the dependencies are installed, run `npm start` to launch the development server
* Boom, your app is ready at `localhost:3000` show your creativity to customize the dashboard as you wish😄

#### All pull requests to better the app are welcome, let's create a dashboard for everyone together with the power of open source🤟🏻


CHANGES.md
# Changes

This is the first time I've contributed to someone elses repo. I came across this app and thought I'd have a go.

I refactored in some places, rearranging some of the useEffect hooks and functions they use.

I removed where the local storage is set to NA and initiate user as null.

## FIX: Adding a theme using the personalize button on the dashboard

This would overwrite the previous themes in the array instead of adding to them.

I fixed (in Personalize.jsx) by setting theme in storage to themeChoice instead of user.theme. This is because the setUser() wouldn't have updated yet and will still hold the previous state.

    setUser({ ...user, theme: themeChoice });
    user.theme  // this won't have the new them choice yet

## Using all image themes

I couldn't see where all user themes were used so I added a random number to pick one from the array each time.

## FIX Style Warning

This was causing a warning in the dev tool console:

    background: `url(${bg})`,

I changed it to this:

    backgroundImage: `url(${bg})`,

I was saying something about using shorthand to set background property. This also seemed to fix a strange re-rendering of background image that was happening.


src/data/Quotes.js
var quotes = [{
        "q": "It's wise to tell the truth. It's even wiser to tell it kindly.",
        "a": "Maxime Lagace",
        "c": "63",
        "h": "<blockquote>&ldquo;It's wise to tell the truth. It's even wiser to tell it kindly.&rdquo; &mdash; <footer>Maxime Lagace</footer></blockquote>"
    },
    {
        "q": "You drown not by falling into a river, but by staying submerged in it.",
        "a": "Paulo Coelho",
        "c": "70",
        "h": "<blockquote>&ldquo;You drown not by falling into a river, but by staying submerged in it.&rdquo; &mdash; <footer>Paulo Coelho</footer></blockquote>"
    },
    {
        "q": "Freeing oneself from words is liberation. ",
        "a": "Bodhidharma",
        "c": "42",
        "h": "<blockquote>&ldquo;Freeing oneself from words is liberation. &rdquo; &mdash; <footer>Bodhidharma</footer></blockquote>"
    },
    {
        "q": "The only way to 'find out if it will work out' is to do it.",
        "a": "Simon Sinek",
        "c": "59",
        "h": "<blockquote>&ldquo;The only way to 'find out if it will work out' is to do it.&rdquo; &mdash; <footer>Simon Sinek</footer></blockquote>"
    },
    {
        "q": "Don't think money does everything or you are going to end up doing everything for money.",
        "a": "Voltaire",
        "c": "88",
        "h": "<blockquote>&ldquo;Don't think money does everything or you are going to end up doing everything for money.&rdquo; &mdash; <footer>Voltaire</footer></blockquote>"
    },
    {
        "q": "Passion creates, addiction consumes.",
        "a": "Gabor Mate",
        "c": "36",
        "h": "<blockquote>&ldquo;Passion creates, addiction consumes.&rdquo; &mdash; <footer>Gabor Mate</footer></blockquote>"
    },
    {
        "q": "Reflect upon your present blessings of which every man has many - not on your past misfortunes, of which all men have some.",
        "a": "Charles Dickens",
        "c": "123",
        "h": "<blockquote>&ldquo;Reflect upon your present blessings of which every man has many - not on your past misfortunes, of which all men have some.&rdquo; &mdash; <footer>Charles Dickens</footer></blockquote>"
    },
    {
        "q": "To the world you may be one person; but to one person you may be the world.",
        "a": "Dr. Seuss",
        "c": "75",
        "h": "<blockquote>&ldquo;To the world you may be one person; but to one person you may be the world.&rdquo; &mdash; <footer>Dr. Seuss</footer></blockquote>"
    },
    {
        "q": "Pleasure in the job puts perfection in the work.",
        "a": "Aristotle",
        "c": "48",
        "h": "<blockquote>&ldquo;Pleasure in the job puts perfection in the work.&rdquo; &mdash; <footer>Aristotle</footer></blockquote>"
    },
    {
        "q": "The decisions of our past are the architects of our present.",
        "a": "Dan Brown",
        "c": "60",
        "h": "<blockquote>&ldquo;The decisions of our past are the architects of our present.&rdquo; &mdash; <footer>Dan Brown</footer></blockquote>"
    },
    {
        "q": "A thing constructed can only be loved after it is constructed; but a thing created is loved before it exists.",
        "a": "Charles Dickens",
        "c": "109",
        "h": "<blockquote>&ldquo;A thing constructed can only be loved after it is constructed; but a thing created is loved before it exists.&rdquo; &mdash; <footer>Charles Dickens</footer></blockquote>"
    },
    {
        "q": "Failure is a prerequisite for great success. If you want success faster, double your rate of failure.",
        "a": "Brian Tracy",
        "c": "101",
        "h": "<blockquote>&ldquo;Failure is a prerequisite for great success. If you want success faster, double your rate of failure.&rdquo; &mdash; <footer>Brian Tracy</footer></blockquote>"
    },
    {
        "q": "You don't need to be better than anyone else, you just need to be better than you used to be.",
        "a": "Wayne Dyer",
        "c": "93",
        "h": "<blockquote>&ldquo;You don't need to be better than anyone else, you just need to be better than you used to be.&rdquo; &mdash; <footer>Wayne Dyer</footer></blockquote>"
    },
    {
        "q": "What we know is a drop, what we don't know is an ocean. ",
        "a": "Isaac Newton",
        "c": "56",
        "h": "<blockquote>&ldquo;What we know is a drop, what we don't know is an ocean. &rdquo; &mdash; <footer>Isaac Newton</footer></blockquote>"
    },
    {
        "q": "We can change our lives. We can do, have, and be exactly what we wish.",
        "a": "Tony Robbins",
        "c": "70",
        "h": "<blockquote>&ldquo;We can change our lives. We can do, have, and be exactly what we wish.&rdquo; &mdash; <footer>Tony Robbins</footer></blockquote>"
    },
    {
        "q": "If you can't imagine it, you can't have it.",
        "a": "Toni Morrison",
        "c": "43",
        "h": "<blockquote>&ldquo;If you can't imagine it, you can't have it.&rdquo; &mdash; <footer>Toni Morrison</footer></blockquote>"
    },
    {
        "q": "You're only given a little spark of madness. You mustn't lose it.",
        "a": "Robin Williams",
        "c": "65",
        "h": "<blockquote>&ldquo;You're only given a little spark of madness. You mustn't lose it.&rdquo; &mdash; <footer>Robin Williams</footer></blockquote>"
    },
    {
        "q": "The whole of life, from the moment you are born to the moment you die, is a process of learning.",
        "a": "Jiddu Krishnamurti",
        "c": "96",
        "h": "<blockquote>&ldquo;The whole of life, from the moment you are born to the moment you die, is a process of learning.&rdquo; &mdash; <footer>Jiddu Krishnamurti</footer></blockquote>"
    },
    {
        "q": "If you put water into a cup, it becomes the cup.",
        "a": "Bruce Lee",
        "c": "48",
        "h": "<blockquote>&ldquo;If you put water into a cup, it becomes the cup.&rdquo; &mdash; <footer>Bruce Lee</footer></blockquote>"
    },
    {
        "q": "Out of suffering have emerged the strongest souls; the most massive characters are seared with scars.",
        "a": "Kahlil Gibran",
        "c": "101",
        "h": "<blockquote>&ldquo;Out of suffering have emerged the strongest souls; the most massive characters are seared with scars.&rdquo; &mdash; <footer>Kahlil Gibran</footer></blockquote>"
    },
    {
        "q": "Only those who will risk going too far can possibly find out how far one can go.",
        "a": "T.S. Eliot",
        "c": "80",
        "h": "<blockquote>&ldquo;Only those who will risk going too far can possibly find out how far one can go.&rdquo; &mdash; <footer>T.S. Eliot</footer></blockquote>"
    },
    {
        "q": "Optimism, when applied to your life, develops strength and peace within you.",
        "a": "Norman Vincent Peale",
        "c": "76",
        "h": "<blockquote>&ldquo;Optimism, when applied to your life, develops strength and peace within you.&rdquo; &mdash; <footer>Norman Vincent Peale</footer></blockquote>"
    },
    {
        "q": "Truth is such a rare thing, it is delighted to tell it.",
        "a": "Emily Dickinson",
        "c": "55",
        "h": "<blockquote>&ldquo;Truth is such a rare thing, it is delighted to tell it.&rdquo; &mdash; <footer>Emily Dickinson</footer></blockquote>"
    },
    {
        "q": "The only thing we have to fear is fear itself.",
        "a": "Franklin D. Roosevelt",
        "c": "46",
        "h": "<blockquote>&ldquo;The only thing we have to fear is fear itself.&rdquo; &mdash; <footer>Franklin D. Roosevelt</footer></blockquote>"
    },
    {
        "q": "It is better to disappoint people with the truth than to appease them with a lie.",
        "a": "Simon Sinek",
        "c": "81",
        "h": "<blockquote>&ldquo;It is better to disappoint people with the truth than to appease them with a lie.&rdquo; &mdash; <footer>Simon Sinek</footer></blockquote>"
    },
    {
        "q": "It will never rain roses: when we want to have more roses, we must plant more roses.",
        "a": "George Eliot",
        "c": "84",
        "h": "<blockquote>&ldquo;It will never rain roses: when we want to have more roses, we must plant more roses.&rdquo; &mdash; <footer>George Eliot</footer></blockquote>"
    },
    {
        "q": "Success is never final, failure is never fatal. It's courage that counts.",
        "a": "John Wooden",
        "c": "73",
        "h": "<blockquote>&ldquo;Success is never final, failure is never fatal. It's courage that counts.&rdquo; &mdash; <footer>John Wooden</footer></blockquote>"
    },
    {
        "q": "When a man is penalized for honesty he learns to lie.",
        "a": "Criss Jami",
        "c": "53",
        "h": "<blockquote>&ldquo;When a man is penalized for honesty he learns to lie.&rdquo; &mdash; <footer>Criss Jami</footer></blockquote>"
    },
    {
        "q": "If we learn to open our hearts, anyone, including the people who drive us crazy, can be our teacher.",
        "a": "Pema Chodron",
        "c": "100",
        "h": "<blockquote>&ldquo;If we learn to open our hearts, anyone, including the people who drive us crazy, can be our teacher.&rdquo; &mdash; <footer>Pema Chodron</footer></blockquote>"
    },
    {
        "q": "A Quitter never wins - and - a Winner never quits.",
        "a": "Napoleon Hill",
        "c": "50",
        "h": "<blockquote>&ldquo;A Quitter never wins - and - a Winner never quits.&rdquo; &mdash; <footer>Napoleon Hill</footer></blockquote>"
    },
    {
        "q": "Begin now to be what you will be hereafter.",
        "a": "St. Jerome",
        "c": "43",
        "h": "<blockquote>&ldquo;Begin now to be what you will be hereafter.&rdquo; &mdash; <footer>St. Jerome</footer></blockquote>"
    },
    {
        "q": "Think how really precious is the time you have to spend, whether it's at work or with your family. Every minute should be enjoyed and savored.",
        "a": "Earl Nightingale",
        "c": "142",
        "h": "<blockquote>&ldquo;Think how really precious is the time you have to spend, whether it's at work or with your family. Every minute should be enjoyed and savored.&rdquo; &mdash; <footer>Earl Nightingale</footer></blockquote>"
    },
    {
        "q": "The destiny of man is in his own soul. ",
        "a": "Herodotus",
        "c": "39",
        "h": "<blockquote>&ldquo;The destiny of man is in his own soul. &rdquo; &mdash; <footer>Herodotus</footer></blockquote>"
    },
    {
        "q": "Learning is a gift, even when pain is your teacher.",
        "a": "Michael Jordan",
        "c": "51",
        "h": "<blockquote>&ldquo;Learning is a gift, even when pain is your teacher.&rdquo; &mdash; <footer>Michael Jordan</footer></blockquote>"
    },
    {
        "q": "Keep smiling, because life is a beautiful thing and there's so much to smile about.",
        "a": "Marilyn Monroe",
        "c": "83",
        "h": "<blockquote>&ldquo;Keep smiling, because life is a beautiful thing and there's so much to smile about.&rdquo; &mdash; <footer>Marilyn Monroe</footer></blockquote>"
    },
    {
        "q": "Failure is not fatal, but failure to change might be.",
        "a": "John Wooden",
        "c": "53",
        "h": "<blockquote>&ldquo;Failure is not fatal, but failure to change might be.&rdquo; &mdash; <footer>John Wooden</footer></blockquote>"
    },
    {
        "q": "Beware of false knowledge, it is more dangerous than ignorance.",
        "a": "George Bernard Shaw",
        "c": "63",
        "h": "<blockquote>&ldquo;Beware of false knowledge, it is more dangerous than ignorance.&rdquo; &mdash; <footer>George Bernard Shaw</footer></blockquote>"
    },
    {
        "q": "Begin at once to live, and count each separate day as a separate life.  ",
        "a": "Seneca",
        "c": "72",
        "h": "<blockquote>&ldquo;Begin at once to live, and count each separate day as a separate life.  &rdquo; &mdash; <footer>Seneca</footer></blockquote>"
    },
    {
        "q": "Lack of emotion causes lack of progress and lack of motivation.",
        "a": "Tony Robbins",
        "c": "63",
        "h": "<blockquote>&ldquo;Lack of emotion causes lack of progress and lack of motivation.&rdquo; &mdash; <footer>Tony Robbins</footer></blockquote>"
    },
    {
        "q": "Believe in one thing too much and you have no room for new ideas.  ",
        "a": "Ray Bradbury",
        "c": "67",
        "h": "<blockquote>&ldquo;Believe in one thing too much and you have no room for new ideas.  &rdquo; &mdash; <footer>Ray Bradbury</footer></blockquote>"
    },
    {
        "q": "It is useless to meet revenge with revenge; it will heal nothing.",
        "a": "J.R.R. Tolkien",
        "c": "65",
        "h": "<blockquote>&ldquo;It is useless to meet revenge with revenge; it will heal nothing.&rdquo; &mdash; <footer>J.R.R. Tolkien</footer></blockquote>"
    },
    {
        "q": "You're always free to change your mind and choose a different future, or a different past.",
        "a": "Richard Bach",
        "c": "90",
        "h": "<blockquote>&ldquo;You're always free to change your mind and choose a different future, or a different past.&rdquo; &mdash; <footer>Richard Bach</footer></blockquote>"
    },
    {
        "q": "Leadership is solving problems",
        "a": "Colin Powell",
        "c": "30",
        "h": "<blockquote>&ldquo;Leadership is solving problems&rdquo; &mdash; <footer>Colin Powell</footer></blockquote>"
    },
    {
        "q": "It is better to fail in originality than to succeed in imitation.",
        "a": "Herman Melville",
        "c": "65",
        "h": "<blockquote>&ldquo;It is better to fail in originality than to succeed in imitation.&rdquo; &mdash; <footer>Herman Melville</footer></blockquote>"
    },
    {
        "q": "Wherever you are is the entry point.",
        "a": "Kabir",
        "c": "36",
        "h": "<blockquote>&ldquo;Wherever you are is the entry point.&rdquo; &mdash; <footer>Kabir</footer></blockquote>"
    },
    {
        "q": "Live calmly. The time will come when the flowers bloom by themselves.",
        "a": "Zen Proverb",
        "c": "69",
        "h": "<blockquote>&ldquo;Live calmly. The time will come when the flowers bloom by themselves.&rdquo; &mdash; <footer>Zen Proverb</footer></blockquote>"
    },
    {
        "q": "If you want to be happy, set a goal that commands your thoughts, liberates your energy, and inspires your hopes.",
        "a": "Andrew Carnegie",
        "c": "112",
        "h": "<blockquote>&ldquo;If you want to be happy, set a goal that commands your thoughts, liberates your energy, and inspires your hopes.&rdquo; &mdash; <footer>Andrew Carnegie</footer></blockquote>"
    },
    {
        "q": "Judge a man by his questions rather than his answers.",
        "a": "Voltaire",
        "c": "53",
        "h": "<blockquote>&ldquo;Judge a man by his questions rather than his answers.&rdquo; &mdash; <footer>Voltaire</footer></blockquote>"
    },
    {
        "q": "Do not ignore the passion that burns in you. Spend time to discover your gift.",
        "a": "Steve Harvey",
        "c": "78",
        "h": "<blockquote>&ldquo;Do not ignore the passion that burns in you. Spend time to discover your gift.&rdquo; &mdash; <footer>Steve Harvey</footer></blockquote>"
    },
    {
        "q": "Ever tried. Ever failed. No matter. Try Again. Fail again. Fail better. ",
        "a": "Samuel Beckett",
        "c": "72",
        "h": "<blockquote>&ldquo;Ever tried. Ever failed. No matter. Try Again. Fail again. Fail better. &rdquo; &mdash; <footer>Samuel Beckett</footer></blockquote>"
    }
]

export default quotes

src/Contexts/UserContext.js
import {
    createContext,
    useEffect,
    useState
} from "react";
import beep from '../sounds/beep-07a.mp3'

export const User = createContext();

const UserContext = ({
    children
}) => {
    //Note state for storing the todos
    const [notes, setNotes] = useState(initNotes())
    //User state to store name, current location, verified status and theme
    const [user, setUser] = useState(initUser())
    // pomodoro


    //To pull tshe notes from localstorage on first loading
    function initNotes() {
        var tasks = localStorage.getItem('notes')
        if (tasks) {
            return JSON.parse(tasks)
        } else
            return []
    }

    //Function to initialize the user if localstoage is empty
    function initUser() {
        return ({
            name: (localStorage.getItem('name')),
            currLocation: (localStorage.getItem('currLocation')),
            verified: (localStorage.getItem('verified')),
            theme: JSON.parse(localStorage.getItem('theme'))
        })
    }
    // state to control the initial time
    const [initialTime, setIntialTime] = useState(0)
    const [timer, setTimer] = useState()
    //  pomodoro timer, function takes in a paramater that is the intial time 
    const startPomodoro = (initialTime) => {
        const timer = setInterval((initialTime) => {
            setIntialTime((initialTime) => initialTime - 1);
            if (initialTime === 0) {
                clearInterval(timer);
            }
        }, 1000);
 
        setTimer(timer);
    }

    // function to play the audio when timer ends
    const audio = new Audio(beep)
    const playAudio = () => {
        const audioPromise = audio.play()
        if (audioPromise !== undefined) {
            audioPromise
                .then(() => {
                    // autoplay started
                    audio.play()
                })
                .catch(err => {
                    // catch dom exception
                    console.info(err)
                })
        }
    }
    useEffect(() => {
        if (initialTime === 0) {
            playAudio()
            clearInterval(timer);
        }
    }, [initialTime, timer]);

    // Run the initializing function on first load
    useEffect(() => {
        initNotes();
        initUser();
    }, [])


    //To update the note to localstorage whenever the note is updated
    useEffect(() => {
        localStorage.setItem('notes', JSON.stringify(notes))
    }, [notes])

    // update localstorage when wallpaper categories are modified
    useEffect(() => {
        // console.log("Context effect fired");
        localStorage.setItem('theme', JSON.stringify(user.theme))
    }, [user.theme])

    return <User.Provider value = {
        {
            user,
            setUser,
            notes,
            setNotes,
            setIntialTime,
            initialTime,
            startPomodoro,
            timer
        }
    } > {
        children
    } < /User.Provider>
}

export default UserContext;

//use effect meant to run on 

src/App.js
import React, { useContext, useState } from "react";
import Greetings from "./Components/Greetings";
import Quote from "./Components/Quote";
import Time from "./Components/Time";
import ToDoMini from "./Components/ToDoMini";
import Weather from "./Components/Weather";
import { FiMenu } from "react-icons/fi";
import { User } from "./Contexts/UserContext";
import Hello from "./Components/Hello";
import Todos from "./Components/Todos";
import { FiSettings } from "react-icons/fi";
import Settings from "./Components/Settings";
import { FaPaintBrush } from "react-icons/fa";
import Personalize from "./Components/Personalize";
import fallBackImage from './data/fallback_bg.jpg'

const App = () => {
  const {user} = useContext(User); //importing the user context which holds the values => name, verified status, theme and current location
  const [todo, setTodo] = useState(false); //Stores the notes
  const [settings, setSettings] = useState(false); // stores toggle state of settings box
  const [personalize, setPersonalize] = useState(false); //stores toggle state of personalize box
  //const [bg, setBg] = useState(fallBackImage)

  //Function to fetch the theme from the localstorage, if found, assign it to bg_image, else assign "nature"
  const bg_image = () => {
    let storageImg = JSON.parse(localStorage.getItem('theme')).toString(); 
    if(!storageImg) return "nature"
    return storageImg
  }

  /*
  const getBg = async() => {
    try {
      const res = await fetch(`https://source.unsplash.com/random/1366x768/?${bg_image}`)
       if(res.url){
         setBg(res.url)
       } 
    } catch (e) {
      console.error(e);
    }
  }
 
  //Calling the function to fetch the theme after the loading of the page is completed
  useEffect(() => {
      getBg();
      // eslint-disable-next-line
    }, [])*/

  return (
    <> 
    {/* Check if the name, location, theme and verified exits in the localstorage aka, if the user is new or existing */}
    {console.log(!user.name || !user.currLocation || !user.verified)}
      {(!user.name || !user.currLocation || !user.verified) ? (
        <Hello /> 
      ) : (
        <div 
          className="main__container h-screen px-5 py-2] bg-opacity-75 bg-cover bg-black"
          style={{
            "background-image": `url(https://source.unsplash.com/random/1366x768/?${bg_image()}), url(${fallBackImage})`, "background-size":'cover' //Image is fetched as per the user's preference
          }}
        >
          <div className="upper_container h-[10%] flex justify-between items-center"> 
          {/* Top container having the todo menu and weather app */}
            <div className="left text-white">
              <button
                className="w-[25vw]"
                onClick={() => {
                  // On clicking the button, toggle the state of the todo menu
                  if (todo) {
                    setTodo(false);
                  } else setTodo(true);
                }} 
              >
                <FiMenu /> {/* The menu icon imported from react-icon */}
              </button>
              {todo && <Todos />} {/* If the toggle state is true, show the todo component */}  
            </div>
            <div className="right">
              <Weather /> {/* Weather component */}
            </div>
          </div>
          <div className="middle_container  h-[80%] flex flex-col justify-center items-center">
          <div className="greetings mb-5">
              <Greetings /> {/* Greeting container */}
            </div>
            <div className="time_conatiner bg-black bg-opacity-30 rounded-3xl px-7">
              <Time /> {/* Time container */}
            </div>
            <div className="todo_main_container bg-black py-1 pb-3 rounded-lg px-3 bg-opacity-40 mt-4">
              <div className="main_tasks text-white text-xl mt-3">
                What are the task for the day
              </div>
              <div className="mini_todo_container">
                <ToDoMini openTodos={() => setTodo(true)} /> {/* Todo input container */}
              </div>
            </div>
            <div className="main_tasks ">
              <Quote /> {/* Quote component */}
            </div>
          </div>
          <div className="lower_container  h-[10%] flex justify-between items-center">
            <div className="setting_div w-[28%] flex flex-col-reverse relative">
              <FiSettings
                style={{ color: "white" }}
                onClick={() => {
                  if (settings) {
                    setSettings(false);
                  } else {
                    setSettings(true);
                  }
                }}
              />
              {settings && <Settings />} {/* Showing the setting component if the toggle state is true */}
            </div>
            <div className="personalize text-white w-[28vw] flex justify-end relative">
            {/* Button to toggle the personalization menu */}
              <button
                className="flex items-center border-2 border-white rounded-full px-3 py-1 bg-black bg-opacity-40"
                onClick={() => {
                  if (personalize) {
                    setPersonalize(false);
                  } else setPersonalize(true);
                }}
              >
                Personalize
                <FaPaintBrush className="mx-2" />
              </button>
              {/* Showing the personalize menu when the toggle state is set to true */}
              {personalize && (
                <Personalize
                  setPersonalize={setPersonalize}
                  personalize={personalize}
                />
              )}
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default App;
